tests/execute/result/custom.sh

```diff
@@ -14,6 +14,7 @@ rlJournalStart
     testName="/test/custom-results"
     rlPhaseStartTest "${testName}"
         rlRun -s "${tmt_command} ${testName} 2>&1 >/dev/null" 1 "Test provides 'results.yaml' file by itself"
+        rlAssertGrep "warn: Result format violation: :3.name - 'without-leading-slash' does not match '^/.*'" $rlRun_LOG
```

Comment:
Doesn't this need `schemas/results.yaml:definitions.subresult.properties.name` to have something like
```yaml
    properties:
      name:
        type: string
        pattern: '^/.*'
```
Or is that pattern checking done outside schema?

Reply:
Hm, I think the answer here might be "no", because, while tests are owned by tmt and tmt knows trees are good and useful and tests are organized in trees, subresults are probably just a one level of nodes in test result subtree, and their names might start as various beakerlib phase names or pytest functions and so on, so not necessarilymatching `^/.*` pattern. I suspect these names will be way more generic. I think we can safely say they may start with `/`, they may contain many `/` to sort of emulate the tree of various subresults, but that's probably not something tmt could/should enforce.

Reply:
Fair, but why the warning in that case though? Could also have it silently not organize the tests results. Otherwise it could be very spammy. I guess this is an issue of controlling warnings, e.g. in CMake there are `-Wdev` and `AUTHOR_WARNING` so that such warnings are only displayed when creating the CMake files/tmt tests.

Reply:
IIUIC, the test checks name of a *result*, that one is supposed to start with a slash, unlike *subresult* name. Result names are then concatenated with the test name, to form `${test.name}${result.name}` results, hence the “starts with a slash” requirement. A result named just `/` represents the result of the test itself.

Reply:
looks resolved to me guys?

Reply:
(the usual issue that github does not allow me to resolve my own comments :upside_down_face:)